examples/dispatch_call_on_tick/main_test.go

```diff
@@ -18,34 +19,55 @@ import (
 )
 
 func TestPluginContext_OnTick(t *testing.T) {
-	opt := proxytest.NewEmulatorOption().WithVMContext(&vmContext{})
-	host, reset := proxytest.NewHostEmulator(opt)
-	defer reset()
-
-	// Call OnVMStart.
-	require.Equal(t, types.OnPluginStartStatusOK, host.StartPlugin())
-	require.Equal(t, tickMilliseconds, host.GetTickPeriod())
-
-	for i := 1; i < 10; i++ {
-		host.Tick() // call OnTick
-		attrs := host.GetCalloutAttributesFromContext(proxytest.PluginContextID)
-		// Verify DispatchHttpCall is called
-		require.Equal(t, len(attrs), i)
-		// Receive callout response.
-		host.CallOnHttpCallResponse(attrs[0].CalloutID, nil, nil, nil)
-		// Check Envoy logs.
-		logs := host.GetInfoLogs()
-		require.Contains(t, logs, fmt.Sprintf("called %d for contextID=%d", i, proxytest.PluginContextID))
-	}
+	vmTest(t, func(vm types.VMContext) {
+		opt := proxytest.NewEmulatorOption().WithVMContext(vm)
+		host, reset := proxytest.NewHostEmulator(opt)
+		defer reset()
 
+		// Call OnVMStart.
+		require.Equal(t, types.OnPluginStartStatusOK, host.StartPlugin())
+		require.Equal(t, tickMilliseconds, host.GetTickPeriod())
+
+		for i := 1; i < 10; i++ {
+			host.Tick() // call OnTick
+			attrs := host.GetCalloutAttributesFromContext(proxytest.PluginContextID)
+			// Verify DispatchHttpCall is called
+			require.Equal(t, len(attrs), i)
+			// Receive callout response.
+			host.CallOnHttpCallResponse(attrs[0].CalloutID, nil, nil, nil)
+			// Check Envoy logs.
+			logs := host.GetInfoLogs()
+			require.Contains(t, logs, fmt.Sprintf("called %d for contextID=%d", i, proxytest.PluginContextID))
+		}
+	})
 }
 
 func TestPluginContext_OnVMStart(t *testing.T) {
-	opt := proxytest.NewEmulatorOption().WithVMContext(&vmContext{})
-	host, reset := proxytest.NewHostEmulator(opt)
-	defer reset()
+	vmTest(t, func(vm types.VMContext) {
+		opt := proxytest.NewEmulatorOption().WithVMContext(vm)
+		host, reset := proxytest.NewHostEmulator(opt)
+		defer reset()
+
+		// Call OnVMStart.
+		require.Equal(t, types.OnPluginStartStatusOK, host.StartPlugin())
+		require.Equal(t, tickMilliseconds, host.GetTickPeriod())
+	})
+}
+
+func vmTest(t *testing.T, f func(types.VMContext)) {
```

Comment:
@mathetake What do you think of this pattern? If OK I'll apply to the remaining folders

Reply:
looks fine, but could you add the util in proxytest pkg?

Reply:
Looked at this but it seemed like it could be awkward or a limited utility. Notably the `t.Skip()` is hard to abstract into a utility. Figured it's not enough code to worry about at least for now

Reply:
oh ok I now I saw this comment